pkg/format/format.go

```diff
@@ -48,9 +49,10 @@ func LogTo(w io.Writer, pod string, container string, output string) {
 
 			if strings.TrimSpace(line) != "" {
 				fields := field.M{
-					"Pod":       pod,
-					"Container": container,
-					"Out":       line,
+					"Pod":             pod,
```

Comment:
I am assuming `LogTo` is the function that gets called to log the things on controller irrespective of the kanister function that is used in the phase.

Reply:
It's currently being used by [`kube.ExecOutput()`](https://github.com/kanisterio/kanister/blob/b5613035af99b47ac277220fbadb7217c15a6af0/pkg/kube/exec.go#L61), which creates the `format.Writer`. So any functions that use `kube.ExecOutput()` gets the new label.

Actually, I should also look into those that calls `kube.Exec()`:

```
$ grep -rniI "kube.exec(" pkg
pkg/restic/restic.go:294:	stdout, stderr, err := kube.Exec(cli, namespace, pod, container, cmd, nil)
pkg/restic/restic.go:322:	stdout, stderr, err := kube.Exec(cli, namespace, pod, container, cmd, nil)
pkg/app/mariadb.go:220:	return kube.Exec(m.cli, m.namespace, podname, containername, command, nil)
pkg/app/cassandra.go:225:	return kube.Exec(cas.cli, cas.namespace, podname, containername, command, nil)
pkg/app/postgresql.go:227:	return kube.Exec(pdb.cli, pdb.namespace, pod, container, command, nil)
pkg/app/elasticsearch.go:223:	return kube.Exec(esi.cli, esi.namespace, podname, containername, command, nil)
pkg/app/foundationdb.go:248:	return kube.Exec(fdb.cli, fdb.namespace, podName, containers[0].Name, command, nil)
pkg/app/mongodb.go:221:	return kube.Exec(mongo.cli, mongo.namespace, podName, containerName, command, nil)
pkg/app/mysql-deploymentconfig.go:198:	stdout, stderr, err := kube.Exec(mdep.cli, mdep.namespace, podName, containerName, command, nil)
pkg/app/postgresql-deploymentconfig.go:221:	return kube.Exec(pgres.cli, pgres.namespace, pod, container, command, nil)
pkg/app/csi-snapshot.go:172:	_, stderr, err := kube.Exec(tlc.cli, tlc.namespace, podname, containername, command, nil)
pkg/app/couchbase.go:262:	return kube.Exec(cb.cli, cb.namespace, podName, container[0].Name, command, nil)
pkg/app/mysql.go:235:	return kube.Exec(mdb.cli, mdb.namespace, podname, containername, command, nil)
pkg/app/mssql.go:248:	return kube.Exec(m.cli, m.namespace, podName, containerName, command, nil)
pkg/app/mongodb-deploymentconfig.go:184:	stdout, stderr, err := kube.Exec(mongo.cli, mongo.namespace, podName, containerName, command, nil)
pkg/function/describe_backups.go:120:		stdout, stderr, err := kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
pkg/function/backup_data.go:167:	stdout, stderr, err := kube.Exec(cli, namespace, pod, container, cmd, nil)
pkg/function/copy_volume_data.go:107:		stdout, stderr, err := kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
pkg/function/restore_data.go:153:		stdout, stderr, err := kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
pkg/function/kube_exec_all.go:103:				stdout, stderr, err := kube.Exec(cli, namespace, p, c, cmd, nil)
pkg/function/backup_data_stats.go:92:		stdout, stderr, err := kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
pkg/function/delete_data.go:101:			stdout, stderr, err := kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
pkg/function/delete_data.go:119:			stdout, stderr, err := kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
pkg/function/delete_data.go:145:	stdout, stderr, err := kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
```

I think they all converge into `format.LogWithCtx()`.

There are a handful of functions (like the RDS ones) that don't use either `kubeTask` or `kubeExec`. If they aren't easy to add, maybe we can just leave them out for now.

Reply:
Looks like `format.LogWithCtx()` is only used for logging data functions; nothing controller-specific.

```sh
$ grep -rniI "logwithctx(" pkg                          
pkg/format/format.go:33:        LogWithCtx(context.Background(), podName, containerName, output)
pkg/format/format.go:86:func LogWithCtx(ctx context.Context, podName string, containerName string, output string) {
pkg/kube/pod_writer.go:48:      format.LogWithCtx(ctx, podName, containerName, stdout)
pkg/kube/pod_writer.go:49:      format.LogWithCtx(ctx, podName, containerName, stderr)
pkg/kube/pod_writer.go:57:      format.LogWithCtx(ctx, podName, containerName, stdout)
pkg/kube/pod_writer.go:58:      format.LogWithCtx(ctx, podName, containerName, stderr)
pkg/function/describe_backups.go:121:           format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stdout)
pkg/function/describe_backups.go:122:           format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stderr)
pkg/function/backup_data.go:168:        format.LogWithCtx(ctx, pod, container, stdout)
pkg/function/backup_data.go:169:        format.LogWithCtx(ctx, pod, container, stderr)
pkg/function/copy_volume_data.go:108:           format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stdout)
pkg/function/copy_volume_data.go:109:           format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stderr)
pkg/function/restore_data.go:154:               format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stdout)
pkg/function/restore_data.go:155:               format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stderr)
pkg/function/kube_exec_all.go:104:                              format.LogWithCtx(ctx, p, c, stdout)
pkg/function/kube_exec_all.go:105:                              format.LogWithCtx(ctx, p, c, stderr)
pkg/function/backup_data_stats.go:93:           format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stdout)
pkg/function/backup_data_stats.go:94:           format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stderr)
pkg/function/prepare_data.go:112:               format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, logs)
pkg/function/delete_data.go:102:                        format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stdout)
pkg/function/delete_data.go:103:                        format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stderr)
pkg/function/delete_data.go:120:                        format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stdout)
pkg/function/delete_data.go:121:                        format.LogWithCtx(ctx, pod.Name, pod.Spec.Containers[0].Name, stderr)
```

Reply:
There was a previous PR to remove commands from logs: https://github.com/kanisterio/kanister/pull/1200, to avoid logging inlined secrets etc.

Reply:
Hi @ihcsim,
The command that you mentioned actually removes `command` field from the pod object that was being dumped in the logs.
I was talking about separate thing, I was talking about the kanister functions that accept command as argument. If output of all those commands are is handled (included in dataPath) here then we are good.

Reply:
I see what you mean. Yeah, all functions the support `command` argument share this log function.